src/community/params-extractor/src/main/java/org/geoserver/params/extractor/rest/EchoesController.java

```diff
@@ -0,0 +1,161 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.params.extractor.rest;
+
+import com.thoughtworks.xstream.XStream;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.geoserver.config.util.XStreamPersister;
+import org.geoserver.params.extractor.EchoParameter;
+import org.geoserver.params.extractor.EchoParameterBuilder;
+import org.geoserver.params.extractor.EchoParameterConverter;
+import org.geoserver.params.extractor.EchoParametersDao;
+import org.geoserver.rest.RequestInfo;
+import org.geoserver.rest.RestBaseController;
+import org.geoserver.rest.RestException;
+import org.geoserver.rest.converters.XStreamMessageConverter;
+import org.geoserver.rest.converters.XStreamXMLMessageConverter;
+import org.geoserver.rest.util.MediaTypeExtensions;
+import org.geoserver.rest.wrapper.RestListWrapper;
+import org.geoserver.rest.wrapper.RestWrapper;
+import org.geoserver.rest.wrapper.RestWrapperAdapter;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Controller to manage the {@link EchoParameter} objects */
+@RestController
+@ControllerAdvice
+@RequestMapping(
+    path = EchoesController.ECHOES_ROOT,
+    produces = {
+        MediaType.APPLICATION_XML_VALUE,
+        MediaType.APPLICATION_JSON_VALUE,
+        MediaTypeExtensions.TEXT_JSON_VALUE
+    }
+)
+public class EchoesController extends RestBaseController {
+    static final String ECHOES_ROOT = RestBaseController.ROOT_PATH + "/params-extractor/echoes";
+
+    @GetMapping
+    public RestListWrapper<EchoParameter> getEchoParameters() {
+        return new RestListWrapper<>(
+                EchoParametersDao.getEchoParameters().stream().collect(Collectors.toList()),
```

Comment:
`EchoParametersDao.getEchoParameters()` already returns a list.

Reply:
It's not going to be encoded by XStream with all the backlinks generation without wrapping in RestListWrapper.

Reply:
Ah, the stream thing.. yes, I was doing map to another type before, and forgot to remove the entire thing.

Reply:
Sorry I was not clear, I was referring to:
```
EchoParametersDao.getEchoParameters().stream().collect(Collectors.toList())
```
... we are converting a list to a list.